pkg/test/expectations/expectations.go

```diff
@@ -223,53 +241,147 @@ func ExpectFinalizersRemoved(ctx context.Context, c client.Client, objs ...clien
 	}
 	// TODO: Check the error on the provisioner scheduling round
 	machines, nodes, _ := provisioner.Schedule(ctx)
-	bindings := map[*v1.Pod]*v1.Node{}
+	bindings := Bindings{}
 	for _, m := range machines {
-		ctx = injection.WithNamespacedName(ctx, types.NamespacedName{Name: m.Labels[v1alpha5.ProvisionerNameLabelKey]})
 		// TODO: Check the error on the provisioner launch
 		name, err := provisioner.Launch(ctx, m, provisioning.WithReason(metrics.ProvisioningReason))
 		if err != nil {
 			return bindings
 		}
-		for _, pod := range m.Pods {
-			bindings[pod] = ExpectNodeExistsWithOffset(offset+1, ctx, c, name)
+		machine := &v1alpha5.Machine{}
+		ExpectWithOffset(offset+1, c.Get(ctx, types.NamespacedName{Name: name}, machine)).To(Succeed())
+		machine, node := ExpectMachineDeployedWithOffset(offset+1, ctx, c, cluster, cloudProvider, machine)
+		if machine != nil && node != nil {
+			for _, pod := range m.Pods {
+				bindings[pod] = &Binding{
+					Machine: machine,
+					Node:    node,
+				}
+			}
 		}
 	}
 	for _, node := range nodes {
 		for _, pod := range node.Pods {
-			bindings[pod] = node.Node.Node
+			bindings[pod] = &Binding{
+				Node:    node.Node,
+				Machine: node.Machine,
+			}
 		}
 	}
 	return bindings
 }
 
+func ExpectMachineDeployed(ctx context.Context, c client.Client, cluster *state.Cluster, cloudProvider cloudprovider.CloudProvider, machine *v1alpha5.Machine) (*v1alpha5.Machine, *v1.Node) {
+	return ExpectMachineDeployedWithOffset(1, ctx, c, cluster, cloudProvider, machine)
+}
+
+func ExpectMachineDeployedWithOffset(offset int, ctx context.Context, c client.Client, cluster *state.Cluster, cloudProvider cloudprovider.CloudProvider, m *v1alpha5.Machine) (*v1alpha5.Machine, *v1.Node) {
+	resolved, err := cloudProvider.Create(ctx, m)
+	// TODO @joinnis: Check this error rather than swallowing it. This is swallowed right now due to how we are doing some testing in the cloudprovider
+	if err != nil {
+		return nil, nil
+	}
+	ExpectWithOffset(offset+1, err).To(Succeed())
+
+	// Make the machine ready in the status conditions
+	machine.PopulateMachineDetails(m, resolved)
+	m.StatusConditions().MarkTrue(v1alpha5.MachineCreated)
+	m.StatusConditions().MarkTrue(v1alpha5.MachineRegistered)
+
+	// Mock the machine launch and node joining at the apiserver
+	node := test.MachineLinkedNode(m)
+	ExpectAppliedWithOffset(offset+1, ctx, c, m, node)
+	ExpectWithOffset(offset+1, cluster.UpdateNode(ctx, node)).To(Succeed())
+	cluster.UpdateMachine(m)
+	return m, node
+}
+
+func ExpectMachinesCascadeDeletion(ctx context.Context, c client.Client, machines ...*v1alpha5.Machine) {
+	nodes := ExpectNodesWithOffset(1, ctx, c)
+	for _, machine := range machines {
+		err := c.Get(ctx, client.ObjectKeyFromObject(machine), &v1alpha5.Machine{})
+		if !errors.IsNotFound(err) {
+			continue
+		}
+		for _, node := range nodes {
+			if node.Spec.ProviderID == machine.Status.ProviderID {
+				Expect(c.Delete(ctx, node))
+			}
+		}
```

Comment:
maybe a lo.Find here as well?
```suggestion
		err := c.Get(ctx, client.ObjectKeyFromObject(machine), &v1alpha5.Machine{})
		if !errors.IsNotFound(err) {
			continue
		}
		if n, found := lo.Find(nodes, func(n *v1.Node, _ int) bool { return n.Spec.ProviderID == machine.Status.ProviderID }); found {
	            Expect(c.Delete(ctx, node))
	         }
		
```

Reply:
No, since we want to delete _every_ node with this providerID, not just the first node